lib/internal/modules/esm/loader.js

```diff
@@ -148,26 +148,28 @@ function nextHookFactory(chain, meta, { validateArgs, validateOutput }) {
   }
 
   return ObjectDefineProperty(
-    async (...args) => {
+    async (arg0, context) => {
       // Update only when hook is invoked to avoid fingering the wrong filePath
       meta.hookErrIdentifier = `${hookFilePath} '${hookName}'`;
 
-      validateArgs(`${meta.hookErrIdentifier} hook's ${nextHookName}()`, args);
+      validateArgs(`${meta.hookErrIdentifier} hook's ${nextHookName}()`, [arg0, context]);
```

Comment:
Let's change the `validateArgs` signature to avoid creating a throw away array (either make it accept 3 argument, or a `{ arg0, context }` object).

Reply:
Just curious: Why is a throw-away object better than a throw-away array here? (I know an array is assigned much more memory, but I think not enough to make a difference)

I chose an array because then the receiver gets to choose the name: resolve's validateArgs names `arg0` as `suppliedSpecifier` and `load's names `arg0` as `nextUrl`. It's _possible_ to rename via object, but kind of messy :/

Reply:
Given that we don't allow array destructuring, it's messy anyway, right? Anyway, you should really pass them as different arguments in this case.

Reply:
My rationale was that that arg encapsulates the args. It just seemed a bit cleaner, but meh 👍